README.md

```diff
@@ -1,6 +1,12 @@
-# needle
+# Needle
 
-Needle is a dependency injection system for Swift iOS apps. It utilizes code generation to ensure compile-time safey. If it compiles, it works.
+Needle is a dependency injection system for Swift iOS apps. Unlike other iOS DI frameworks, such as [Cleanse](https://github.com/square/Cleanse), [Swinject](https://github.com/Swinject/Swinject), Needle encourages **hierarchical DI structure and utilizes code generation to ensure compile-time safey**. This allows us to develop our apps and making changes with confidence. **If it compiles, it works.** In this aspect, Needle is more similar to [Android Dagger](https://google.github.io/dagger/).
+
+## [Why use dependency injection?](https://github.com/uber/needle/WHY_DI.md)
+
+The linked document uses a somewhat real iOS example to explain what the dependency injection pattern is, and its benefits.
```

Comment:
nit: drop 'iOS' maybe ?

Reply:
It mentions view controllers and stuff, which I thought was iOS specific. Or are you saying it's redundant since Needle is for iOS?

Reply:
I meant it applies to the desktop as well, you never mention `UIViewController` so your comments apply to `NSViewControllers` as well.

Reply:
Haha. Sure sounds good. We've never planned this for anything other than iOS. Although I suppose given Apple is pushing Swift for more than just iOS, we don't have to be specific here either.